src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -503,7 +503,7 @@ static TypeSymbol RemoveImplicitNull(TypeSymbol type, bool typeWasPreserved)
                         }
                         else
                         {
-                            diagnosticWriter.Write(config.OriginSyntax ?? declaredPropertySyntax.Key, x => x.CannotAssignToReadOnlyProperty(ShouldWarn(targetType), declaredProperty.Name));
```

Comment:
Similar to the [`DisallowedProperty` diagnostic](https://github.com/Azure/bicep/blob/71f6ce7bafc3ffc1137f0db66de1a263beb16269/src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs#L272), shouldn't we check whether the readonly property assignment is on a resource type, and only use warning + show the `TypeInaccuracyLink` if it's a resource?

For example, we'd want to continue blocking assignment to `outputs` on a module.

Reply:
Fixed. I also excluded showing type inaccuracy link for id, type and api-version properties.